ethereum/contracts/relayer/wormholeRelayer/WormholeRelayerSerde.sol

```diff
@@ -0,0 +1,239 @@
+
+        strct.requestedReceiverValue = TargetNative.wrap(requestedReceiverValue);
+        strct.extraReceiverValue = TargetNative.wrap(extraReceiverValue);
+
+        checkLength(encoded, offset);
+    }
+
+    function encode(RedeliveryInstruction memory strct)
+        internal
+        pure
+        returns (bytes memory encoded)
+    {
+        bytes memory vaaKey = encodeVaaKey(strct.deliveryVaaKey);
+        encoded = abi.encodePacked(
+            PAYLOAD_ID_REDELIVERY_INSTRUCTION,
+            vaaKey,
+            strct.targetChain,
+            strct.newRequestedReceiverValue,
+            encodeBytes(strct.newEncodedExecutionInfo),
+            strct.newSourceDeliveryProvider,
+            strct.newSenderAddress
+        );
+    }
+
+    function decodeRedeliveryInstruction(bytes memory encoded)
+        internal
+        pure
+        returns (RedeliveryInstruction memory strct)
+    {
+        uint256 offset = checkUint8(encoded, 0, PAYLOAD_ID_REDELIVERY_INSTRUCTION);
+
+        uint256 newRequestedReceiverValue;
+
+        (strct.deliveryVaaKey, offset) = decodeVaaKey(encoded, offset);
+        (strct.targetChain, offset) = encoded.asUint16Unchecked(offset);
+        (newRequestedReceiverValue, offset) = encoded.asUint256Unchecked(offset);
+        (strct.newEncodedExecutionInfo, offset) = decodeBytes(encoded, offset);
+        (strct.newSourceDeliveryProvider, offset) = encoded.asBytes32Unchecked(offset);
+        (strct.newSenderAddress, offset) = encoded.asBytes32Unchecked(offset);
+
+        strct.newRequestedReceiverValue = TargetNative.wrap(newRequestedReceiverValue);
+
+        checkLength(encoded, offset);
+    }
+
+    function encode(DeliveryOverride memory strct) internal pure returns (bytes memory encoded) {
+        encoded = abi.encodePacked(
+            VERSION_DELIVERY_OVERRIDE,
+            strct.newReceiverValue,
+            encodeBytes(strct.newExecutionInfo),
+            strct.redeliveryHash
+        );
+    }
+
+    function decodeDeliveryOverride(bytes memory encoded)
+        internal
+        pure
+        returns (DeliveryOverride memory strct)
+    {
+        uint256 offset = checkUint8(encoded, 0, VERSION_DELIVERY_OVERRIDE);
+
+        uint256 receiverValue;
+
+        (receiverValue, offset) = encoded.asUint256Unchecked(offset);
+        (strct.newExecutionInfo, offset) = decodeBytes(encoded, offset);
+        (strct.redeliveryHash, offset) = encoded.asBytes32Unchecked(offset);
+
+        strct.newReceiverValue = TargetNative.wrap(receiverValue);
+
+        checkLength(encoded, offset);
+    }
+
+    // ------------------------------------------ private --------------------------------------------
+
+    function encodeVaaKeyArray(VaaKey[] memory vaaKeys)
+        private
+        pure
+        returns (bytes memory encoded)
+    {
+        assert(vaaKeys.length < type(uint8).max);
+        encoded = abi.encodePacked(uint8(vaaKeys.length));
+        for (uint256 i = 0; i < vaaKeys.length;) {
```

Comment:
We should cache the `vaaKeys.length` variable to make sure we're not calling `.length` on every iteration.